portfolio/src/main/webapp/scripts/server-dev.js

```diff
@@ -12,18 +12,41 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
 /**
  * Fetches the previously entered comments from the server and inserts each
  * comment as a list item of the 'comments' <ul> element.
+ *
+ * An option to determine the maximum number of comments is also included
+ * using a Query String parameter created from the num-comments form. When
+ * the page is (re-)loaded, the number of comments displayed is determined
+ * from the selection in the previous session. Otherwise, the most recently
+ * submitted number selection will be used. The number of comments will
+ * also never exceed the number of total comments returned from the datastore.
  */
 function getCommentsThread() {
   fetch('/comment-data')
       .then(response => response.json())
       .then((commentList) => {
         const commentThread = document.getElementById('comments-thread');
-        commentList.forEach((comment) => {
-          commentThread.appendChild(createListElement(comment));
-        });
+        const urlParams = new URLSearchParams(window.location.search);
+
+        // Determine the number of comments to display.
+        let numComments = urlParams.get('num-comments');
+        const numCommentsStored = parseInt(
+            sessionStorage.getItem('numComments'));
+        if (numComments == null) {
+          numComments = numCommentsStored;
+        } else {
+          sessionStorage.setItem('numComments', numComments);
+        }
+        const maxCommentIdx = Math.min(numComments, commentList.length);
```

Comment:
is this a real ID or more like a maxNumOfComment? If you switch this line with line 44, you can reuse the numComments variable, for this maxCommentIdx user case, which makes more sense logically to me.

Reply:
I just carried forward the changes made from Hiram's comments/suggestions in #16. Please let me know if there is still confusion/suggestions after reading the updated version.